text/0005-white-flag/0005-white-flag.md

```diff
@@ -0,0 +1,54 @@
++ Feature name: `white-flag`
++ Start date: 2020-03-06
++ RFC PR: [iotaledger/protocol-rfcs#0005](https://github.com/iotaledger/protocol-rfcs/pull/5)
+
+# Summary
+
+Allow milestones to confirm conflicting bundles. Mutate the ledger state by deterministically ordering the bundles and
+ignoring conflicts.
+
+# Motivation
+
+The main motivations:
+
+- Defend against censorship attacks - Conflicts will no longer block bundles from being approved.
+- Make reattachments unnecessary - As long as the network is not saturated, theoretically all bundles should be
+approved. And no bundle will be left behind.
+- Increase TPS - Due to easy node tip selection the network throughput should increase.
+- Increase CTPS - Due to the above, increase in TPS and no left-behinds, we expect CTPS to increase as well.
+
+# Detailed design
+
+Let's define a conflicting bundle as a bundle that leads to a negative balance on an address if applied to the current
+ledger state.
+
+When a milestone is accepted by a node then the following algorithm to mutate the ledger state is performed:
+
+- Look for milestone whose index is the current latest solid milestone index + 1. Milestones must be applied in order!
+- When the milestone is found start traversing down the tangle in DFS going first to the trunk and only then the branch.
+Stop at the transactions that has only approved parents.
+- Start applying bundles to the diffmap only as you go up the recursion stack (when you climb up back from the leafs).
+- Every time you attempt to apply a bundle to the diffmap:
+If it is invalid, mark it as seen and ignored by the milestone.
+If it is valid and not conflicting with the current state then apply it to the ledger state. Mark it as seen and
+approved.
```

Comment:
approved or confirmed?
or maybe a different word because either one it eventually means that the mentioned bundle spent address (and spent into) are deducted/added in the ledger state